auth.go

```diff
@@ -28,31 +32,85 @@ func pathLogin(b *mesosBackend) *framework.Path {
 
 // pathLogin (the method) is the "login" path request handler.
 func (b *mesosBackend) pathLogin(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-
 	taskID := d.Get("task-id").(string)
-	if len(taskID) < 1 {
+	if !b.verifyTaskExists(taskID) {
+		return nil, logical.ErrPermissionDenied
+	}
+
+	prefix, err := taskIDPrefix(taskID)
+	if err != nil {
 		return nil, logical.ErrPermissionDenied
 	}
 
-	b.Logger().Info("LOGIN", "task-id", taskID, "RemoteAddr", req.Connection.RemoteAddr)
+	b.Logger().Info("LOGIN", "task-id", taskID, "prefix", prefix, "RemoteAddr", req.Connection.RemoteAddr)
+
+	policies, err := getTaskPolicies(ctx, req.Storage, prefix)
+	if err != nil {
+		return nil, err
+	}
 
-	// TODO: Validate the task-id and look up the associated list of policies.
 	// TODO: Make the renewal period configurable?
 	return &logical.Response{
 		Auth: &logical.Auth{
-			Policies: []string{},
+			Policies: policies,
 			Period:   10 * time.Minute,
 			LeaseOptions: logical.LeaseOptions{
 				Renewable: true,
 			},
+			// Stash task-id so we can check it again for renewals.
+			InternalData: jsonobj{"task-id": taskID},
 		},
 	}, nil
 }
 
 // authRenew is the renew callback for tokens created by this plugin.
 func (b *mesosBackend) authRenew(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	b.Logger().Info("RENEW", "req", fmt.Sprintf("%#v", req))
-	// For an unconditional renewal, we only need to return the Auth struct
+	b.Logger().Info("RENEW", "auth", fmt.Sprintf("%#v", req.Auth))
+
+	taskID := req.Auth.InternalData["task-id"].(string)
```

Comment:
Do we have the taskID that was sent in the auth renewal request, if there is one?

Reply:
The only place we get the `taskID` is the login call, so we have to stash it in `Auth.InternalData` to get access to it in renewals.